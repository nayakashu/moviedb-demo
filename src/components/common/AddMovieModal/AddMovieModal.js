import React, { Component } from 'react';
import { connect } from 'react-redux';
import { closeLoginForm } from '../../../actions/modalActions';
import './AddMovieModal.scss';
import makeModal from '../../hocs/makeModal/makeModal';

export class AddMovieModal extends Component {
  render() {
    return (
      <div className="login-form">
        <span className="logo">
          <i className="material-icons">timeline</i>
          <span className="logo-text">Add Movie</span>
        </span>

        <i
          className="material-icons btn-close"
          onClick={this.props.closeLoginForm}
        >
          close
        </i>

        <form>
          <input type="text" placeholder="Movie Title" />
          <input type="text" placeholder="Casts" />
          <div className="row">
            <div className="col-sm-6 icon-overlay">
              <input type="text" placeholder="Date of Launch" />
              <i className="material-icons">date_range</i>
            </div>
            <div className="col-sm-6 icon-overlay">
              <input type="text" placeholder="Movie Poster" />
              <i className="material-icons">cloud_upload</i>
            </div>
          </div>
          <textarea className="txt-movie-desc" placeholder="Movie overview" />
          <div className="row">
            <div className="col-sm-6">
              <button className="btn-cancel">Cancel</button>
            </div>
            <div className="col-sm-6">
              <button className="btn-login">Add Movie</button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export default makeModal(
  connect(
    null,
    { closeLoginForm }
  )(AddMovieModal)
);
