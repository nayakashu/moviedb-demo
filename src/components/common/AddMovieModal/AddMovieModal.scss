@import '../../../Common.scss';
.row {
  .icon-overlay {
    i {
      position: absolute;
      color: #fff;
      right: 39px;
      top: 19px;
      color: $content-text-color;
    }
  }

  .col-sm-6 {
    button.btn-cancel {
      background: transparent;
      border: none;
      width: 100%;
      padding: 20px;
      color: $content-text-color;
      outline: none;
    }
  }
}

.txt-movie-desc {
  width: 100%;
  background: $body-bg-color;
  border: none;
  padding: 20px;
  margin-bottom: 25px;
  height: 150px;
  outline: none;
  color: $content-text-color;
}
