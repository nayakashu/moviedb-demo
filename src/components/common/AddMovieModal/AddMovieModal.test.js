import React from 'react';
import { shallow } from 'enzyme';
import { AddMovieModal } from './AddMovieModal';

describe('AddMovieModal', () => {
  const addMovieModal = shallow(<AddMovieModal />);

  it('renders correctly', () => {
    expect(addMovieModal).toMatchSnapshot();
  });
});
