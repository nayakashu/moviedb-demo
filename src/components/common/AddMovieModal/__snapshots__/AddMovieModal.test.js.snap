// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieModal renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddMovieModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="logo"
        >
          <i
            className="material-icons"
          >
            timeline
          </i>
          <span
            className="logo-text"
          >
            Add Movie
          </span>
        </span>,
        <i
          className="material-icons btn-close"
        >
          close
        </i>,
        <form>
          <input
            placeholder="Movie Title"
            type="text"
          />
          <input
            placeholder="Casts"
            type="text"
          />
          <div
            className="row"
          >
            <div
              className="col-sm-6 icon-overlay"
            >
              <input
                placeholder="Date of Launch"
                type="text"
              />
              <i
                className="material-icons"
              >
                date_range
              </i>
            </div>
            <div
              className="col-sm-6 icon-overlay"
            >
              <input
                placeholder="Movie Poster"
                type="text"
              />
              <i
                className="material-icons"
              >
                cloud_upload
              </i>
            </div>
          </div>
          <textarea
            className="txt-movie-desc"
            placeholder="Movie overview"
          />
          <div
            className="row"
          >
            <div
              className="col-sm-6"
            >
              <button
                className="btn-cancel"
              >
                Cancel
              </button>
            </div>
            <div
              className="col-sm-6"
            >
              <button
                className="btn-login"
                onClick={[Function]}
              >
                Add Movie
              </button>
            </div>
          </div>
        </form>,
      ],
      "className": "login-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="material-icons"
            >
              timeline
            </i>,
            <span
              className="logo-text"
            >
              Add Movie
            </span>,
          ],
          "className": "logo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "timeline",
              "className": "material-icons",
            },
            "ref": null,
            "rendered": "timeline",
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Movie",
              "className": "logo-text",
            },
            "ref": null,
            "rendered": "Add Movie",
            "type": "span",
          },
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "close",
          "className": "material-icons btn-close",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "close",
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              placeholder="Movie Title"
              type="text"
            />,
            <input
              placeholder="Casts"
              type="text"
            />,
            <div
              className="row"
            >
              <div
                className="col-sm-6 icon-overlay"
              >
                <input
                  placeholder="Date of Launch"
                  type="text"
                />
                <i
                  className="material-icons"
                >
                  date_range
                </i>
              </div>
              <div
                className="col-sm-6 icon-overlay"
              >
                <input
                  placeholder="Movie Poster"
                  type="text"
                />
                <i
                  className="material-icons"
                >
                  cloud_upload
                </i>
              </div>
            </div>,
            <textarea
              className="txt-movie-desc"
              placeholder="Movie overview"
            />,
            <div
              className="row"
            >
              <div
                className="col-sm-6"
              >
                <button
                  className="btn-cancel"
                >
                  Cancel
                </button>
              </div>
              <div
                className="col-sm-6"
              >
                <button
                  className="btn-login"
                  onClick={[Function]}
                >
                  Add Movie
                </button>
              </div>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "placeholder": "Movie Title",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "placeholder": "Casts",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-sm-6 icon-overlay"
                >
                  <input
                    placeholder="Date of Launch"
                    type="text"
                  />
                  <i
                    className="material-icons"
                  >
                    date_range
                  </i>
                </div>,
                <div
                  className="col-sm-6 icon-overlay"
                >
                  <input
                    placeholder="Movie Poster"
                    type="text"
                  />
                  <i
                    className="material-icons"
                  >
                    cloud_upload
                  </i>
                </div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      placeholder="Date of Launch"
                      type="text"
                    />,
                    <i
                      className="material-icons"
                    >
                      date_range
                    </i>,
                  ],
                  "className": "col-sm-6 icon-overlay",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "placeholder": "Date of Launch",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "date_range",
                      "className": "material-icons",
                    },
                    "ref": null,
                    "rendered": "date_range",
                    "type": "i",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      placeholder="Movie Poster"
                      type="text"
                    />,
                    <i
                      className="material-icons"
                    >
                      cloud_upload
                    </i>,
                  ],
                  "className": "col-sm-6 icon-overlay",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "placeholder": "Movie Poster",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "cloud_upload",
                      "className": "material-icons",
                    },
                    "ref": null,
                    "rendered": "cloud_upload",
                    "type": "i",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "txt-movie-desc",
              "placeholder": "Movie overview",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-sm-6"
                >
                  <button
                    className="btn-cancel"
                  >
                    Cancel
                  </button>
                </div>,
                <div
                  className="col-sm-6"
                >
                  <button
                    className="btn-login"
                    onClick={[Function]}
                  >
                    Add Movie
                  </button>
                </div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn-cancel"
                  >
                    Cancel
                  </button>,
                  "className": "col-sm-6",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cancel",
                    "className": "btn-cancel",
                    "onClick": undefined,
                  },
                  "ref": null,
                  "rendered": "Cancel",
                  "type": "button",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn-login"
                    onClick={[Function]}
                  >
                    Add Movie
                  </button>,
                  "className": "col-sm-6",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add Movie",
                    "className": "btn-login",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Add Movie",
                  "type": "button",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="logo"
          >
            <i
              className="material-icons"
            >
              timeline
            </i>
            <span
              className="logo-text"
            >
              Add Movie
            </span>
          </span>,
          <i
            className="material-icons btn-close"
          >
            close
          </i>,
          <form>
            <input
              placeholder="Movie Title"
              type="text"
            />
            <input
              placeholder="Casts"
              type="text"
            />
            <div
              className="row"
            >
              <div
                className="col-sm-6 icon-overlay"
              >
                <input
                  placeholder="Date of Launch"
                  type="text"
                />
                <i
                  className="material-icons"
                >
                  date_range
                </i>
              </div>
              <div
                className="col-sm-6 icon-overlay"
              >
                <input
                  placeholder="Movie Poster"
                  type="text"
                />
                <i
                  className="material-icons"
                >
                  cloud_upload
                </i>
              </div>
            </div>
            <textarea
              className="txt-movie-desc"
              placeholder="Movie overview"
            />
            <div
              className="row"
            >
              <div
                className="col-sm-6"
              >
                <button
                  className="btn-cancel"
                >
                  Cancel
                </button>
              </div>
              <div
                className="col-sm-6"
              >
                <button
                  className="btn-login"
                  onClick={[Function]}
                >
                  Add Movie
                </button>
              </div>
            </div>
          </form>,
        ],
        "className": "login-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="material-icons"
              >
                timeline
              </i>,
              <span
                className="logo-text"
              >
                Add Movie
              </span>,
            ],
            "className": "logo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "timeline",
                "className": "material-icons",
              },
              "ref": null,
              "rendered": "timeline",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Movie",
                "className": "logo-text",
              },
              "ref": null,
              "rendered": "Add Movie",
              "type": "span",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "close",
            "className": "material-icons btn-close",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "close",
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                placeholder="Movie Title"
                type="text"
              />,
              <input
                placeholder="Casts"
                type="text"
              />,
              <div
                className="row"
              >
                <div
                  className="col-sm-6 icon-overlay"
                >
                  <input
                    placeholder="Date of Launch"
                    type="text"
                  />
                  <i
                    className="material-icons"
                  >
                    date_range
                  </i>
                </div>
                <div
                  className="col-sm-6 icon-overlay"
                >
                  <input
                    placeholder="Movie Poster"
                    type="text"
                  />
                  <i
                    className="material-icons"
                  >
                    cloud_upload
                  </i>
                </div>
              </div>,
              <textarea
                className="txt-movie-desc"
                placeholder="Movie overview"
              />,
              <div
                className="row"
              >
                <div
                  className="col-sm-6"
                >
                  <button
                    className="btn-cancel"
                  >
                    Cancel
                  </button>
                </div>
                <div
                  className="col-sm-6"
                >
                  <button
                    className="btn-login"
                    onClick={[Function]}
                  >
                    Add Movie
                  </button>
                </div>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "Movie Title",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "Casts",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="col-sm-6 icon-overlay"
                  >
                    <input
                      placeholder="Date of Launch"
                      type="text"
                    />
                    <i
                      className="material-icons"
                    >
                      date_range
                    </i>
                  </div>,
                  <div
                    className="col-sm-6 icon-overlay"
                  >
                    <input
                      placeholder="Movie Poster"
                      type="text"
                    />
                    <i
                      className="material-icons"
                    >
                      cloud_upload
                    </i>
                  </div>,
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        placeholder="Date of Launch"
                        type="text"
                      />,
                      <i
                        className="material-icons"
                      >
                        date_range
                      </i>,
                    ],
                    "className": "col-sm-6 icon-overlay",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "placeholder": "Date of Launch",
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "date_range",
                        "className": "material-icons",
                      },
                      "ref": null,
                      "rendered": "date_range",
                      "type": "i",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        placeholder="Movie Poster"
                        type="text"
                      />,
                      <i
                        className="material-icons"
                      >
                        cloud_upload
                      </i>,
                    ],
                    "className": "col-sm-6 icon-overlay",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "placeholder": "Movie Poster",
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "cloud_upload",
                        "className": "material-icons",
                      },
                      "ref": null,
                      "rendered": "cloud_upload",
                      "type": "i",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "txt-movie-desc",
                "placeholder": "Movie overview",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="col-sm-6"
                  >
                    <button
                      className="btn-cancel"
                    >
                      Cancel
                    </button>
                  </div>,
                  <div
                    className="col-sm-6"
                  >
                    <button
                      className="btn-login"
                      onClick={[Function]}
                    >
                      Add Movie
                    </button>
                  </div>,
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="btn-cancel"
                    >
                      Cancel
                    </button>,
                    "className": "col-sm-6",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Cancel",
                      "className": "btn-cancel",
                      "onClick": undefined,
                    },
                    "ref": null,
                    "rendered": "Cancel",
                    "type": "button",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="btn-login"
                      onClick={[Function]}
                    >
                      Add Movie
                    </button>,
                    "className": "col-sm-6",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Add Movie",
                      "className": "btn-login",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Add Movie",
                    "type": "button",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
